service-runner.js

```diff
@@ -105,44 +100,103 @@ ServiceRunner.prototype._sanitizeConfig = function(conf, options) {
     return conf;
 };
 
-ServiceRunner.prototype.updateConfig = function updateConfig(conf) {
+
+/**
+ * Loads the config from file, serialized input or Object
+ *
+ * @param conf a configuration.
+ *             If undefined - config is loaded from config.yaml file
+ *             If Object - treated as already parsed configuration
+ *             If sting - treated a serialized yaml config
+ */
+ServiceRunner.prototype.loadConfig = function loadConfig(conf) {
     var self = this;
-    if (conf) {
+    var action;
+    if (conf && conf instanceof Object) {
```

Comment:
Hm, no, indeed this case is a bit tricky, since it doesn't need parsing, but we'd still want to load the package info we add below.